osv/cache.py

```diff
@@ -56,3 +59,30 @@ def get(self, key):
 
   def set(self, key, value, ttl):
     self.key_val_map[key] = _CacheEntry(value, ttl)
+
+
+def Cached(cache: Cache, ttl: int = 60 * 60, cache_blacklist=frozenset()):
+  """
+  Decorates your function to cache results
+  :param cache_blacklist: Avoid caching these results
+  :param cache: Cache object to store information
+  :param ttl: Time to live in seconds (default 1 hour)
+  """
+
+  def decorator(func):
+
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs):
+      key = frozenset(getcallargs(func, *args, **kwargs).items())
```

Comment:
I think we should include the function module + name here too. It seems pretty likely that one may re-use the same cache object for multiple functions.

Reply:
I thought that was the benefit of passing in the cache object, so that the caller can control what sharing of cache they want to have... So if they pass the same cache object into two functions, we assume that's intentional?

Reply:
That seems like it would be more surprising than not. Less magic is always better than more, and if something truly needs to be shared between two different functions, then that should be refactored out.

It also makes it difficult to e.g. share a single redis cache across multiple functions. There's no easy mechanism to use the same redis instance for multiple functions right now.

Reply:
Ah yeah that makes sense